packages/vite/src/node/preview.ts

```diff
@@ -52,8 +52,11 @@ export async function preview(
   const protocol = options.https ? 'https' : 'http'
   const logger = config.logger
   const base = config.base
+  const socket = options.socket
 
-  httpServer.listen(port, hostname.host, () => {
+  const listenOption = socket ? socket : { port, host: hostname }
```

Comment:
This can be shortcutted to:

```suggestion
  const listenOption = socket ?? { port, host: hostname }
```